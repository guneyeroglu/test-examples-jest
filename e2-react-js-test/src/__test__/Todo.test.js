import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Todo from '../components/Todo';

afterEach(() => cleanup());

test('shoulds render non-completed todo', () => {
  const todo = { id: 1, title: 'wash dishes', completed: false };
  render(<Todo {...todo} />);
  //
  const todoElement = screen.getByTestId(`todo-${todo.id}`);
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent(/wash dishes/i);
  expect(todoElement).not.toContainHTML('<strike>');
});

test('shoulds render completed todo', () => {
  const todo = { id: 2, title: 'wash car', completed: true };
  render(<Todo {...todo} />);
  //
  const todoElement = screen.getByTestId(`todo-${todo.id}`);
  expect(todoElement).toBeInTheDocument();
  expect(todoElement).toHaveTextContent(/wash car/i);
  expect(todoElement).not.toContainHTML('<strike>');
});

test('matches snapshot', () => {
  const todo = { id: 1, title: 'wash dishes', completed: false };
  const tree = renderer.create(<Todo {...todo} />).toJSON();
  //
  console.log(tree);
  /*
    type: ...,
    props: ...,
    children: ...,
  */
  //

  expect(tree).toMatchSnapshot();
});
