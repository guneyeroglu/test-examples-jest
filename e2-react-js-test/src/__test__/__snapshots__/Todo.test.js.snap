// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="todo-3"
>
  <h1>
    make dinner
  </h1>
</div>
`;
